pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -222,6 +184,12 @@ func postProcessRegistries(regs []Registry) ([]Registry, error) {
 			}
 		}
 
+		// make sure a Prefix ends with "/" to ensure matching at
+		// path and hence at namespace boundaries
+		if !strings.HasSuffix(reg.Prefix, "/") {
```

Comment:
Couldn't this be just moved to the check in `FindRegistry` instead of leaking into user-visible data?

Also, AFAICS this breaks the `CheckAuth` case (`FindRegistry(…, hostnameWithoutAnySlashes)`).

Reply:
… sure, independently it’s perfectly reasonable to define that canonical `Registry.Prefix` values always end with exactly one slash, and this is just normalizes the values to confirm to this definition, but if `FindRegistry`, the primary user, ends up having to undo that, then it’s a bit impractical.

Reply:
> Also, AFAICS this breaks the CheckAuth case (FindRegistry(…, hostnameWithoutAnySlashes)).

I think that `hostnameWithoutAnySlashes` should be appended with a "/" in `FindRegistry(...)`? I didn't (yet) add code to throw an error if a Prefix doesn't end with slash but likely that's going to happen (to make it more transparent to the user). Once that happens, `hostnameWithoutAnySlashes` is problematic.

Reply:
> I think that `hostnameWithoutAnySlashes` should be appended with a "/" in `FindRegistry(...)`?

Would that involve adding an extra check whether the input is a hostname without any slashes, or would a slash be unconditionally added (even if the input is a named ref, possibly ending with a tag or digest)?

Even to implement precisely the proposed semantics (`.Prefix` matches hostnames and namespaces, not repos or tagged/digested images — at this point I’m fine with that, I’m _not_ arguing about what the semantics should be, except that `CheckAuth` should work), adding the trailing slash to the `.Prefix` value just seems to make things more complex instead of simpler.

Or, of course, I’m being stupid — I haven’t actually tested anything about this in practice, so I may well be misunderstanding how the code actually works.

Reply:
No at all. You have a very valid point. Then let's go with your initial proposal (similar to `strings.HasPrefix(ref, r.Prefix + "/") || ref == r.Prefix`).

Reply:
Pushed the changes.